docs/src/tutorials/Add-New-Network.md

```diff
@@ -0,0 +1,164 @@
+# Add New Network
+
+Parity Signer supports adding any substrate-based networks or updating and existing network via QR code.
+
+After you've installed [required software](#Prerequisites), you need to add *Network's Specs* to Signer and add *Network Metadata* for this network, so that Signer is able to decode, and you could read and verify transactions you are signing on this network.
+
+If you need to update metadata for already existing network you only need to update *Network Metadata*.
+Off-the-shelf Signer comes with networks that you can update by scanning a multipart QR codes that containt recent metadata for these networks at [Metadata Update Portal](https://metadata.parity.io/).
+
+*Network Specs*
+
+1. [Get](#get-network-spec)
+2. [Sign](#sign-network-spec)
+3. [Feed into Signer](#feed-network-spec-into-signer)
+
+*Network Metadata*
+
+4. [Get](#get-network-metadata)
+5. [Sign](#sign-network-metadata)
+6. [Feed into Signer](#feed-network-metadata-into-signer)
+
+### Prerequisites
+
+- [ ] Websocket URL of the network (we maintain an [address book](https://github.com/paritytech/parity-signer/blob/master/rust/generate_message/address_book))
+- [ ] [rust](https://www.rust-lang.org/tools/install)
+- [ ] [subkey](https://docs.substrate.io/v3/tools/subkey/#installation)
+- [ ] a clone of [parity-signer repository](https://github.com/paritytech/parity-signer). In the repository go to `generate_message` crate:
+
+```
+cd rust/generate_message
+cargo build
+cargo run restore_defaults
+```
+
+With `subkey` you need to generate a private and public keys (`<secret-phrase>` and `Public key (hex)`). You will be able to update a network only with metadata that is signed with the same keys as network specs that defined the network for your Signer instance.
+
+```
+subkey generate
+```
+
+---
+
+## Network Specs
+
+### Get Network Specs
+
+In `parity-signer/rust/generate_message`
+
+```
+cargo run add_spec -u <network-ws-url> -<crypto>
+
+```
+```
+// i.g.
+cargo run add_specs -u wss://statemint-rpc.polkadot.io -sr25519
+
+```
+
+For networks supporting several tokens:
+
+```
+cargo run add_spec -u <network-ws-url> -<crypto> -token <decimals> <SYMBOL>
+
+```
+```
+// i.g.
+cargo run add_spec -u wss://karura-rpc-0.aca-api.network -sr25519 -token 12 KAR
+
+```
+
+Now your `<specs-file>` is in `parity-signer/rust/files/for_signing`
+
+
+### Sign Network Spec
+
+#### Get signature
+
+In `parity-signer/rust/files/for_signing`
+
+```
+cat <spec-file> | subkey sign --suri <seed-phrase>
+```
+
+```
+// i.g.
+cat sign_me_add_specs_statemint_sr25519 | subkey sign --suri "fluid upon nuclear level middle funny suffer author group shrimp tornado relax"
+```
+
+This will return a `<signature>` you need to make a signed QR.
+
+#### Make signed QR
+
+In `parity-signer/rust/generate_message`
+
+```
+cargo run make -qr -crypto <crypto> -msgtype add_specs -payload <spec-file> -verifier -hex <signature>
```

Comment:
This is quite slow, unless we add `--release` to this command. It would be nice to add `--release` to all these calls, especially the slow ones:

`cargo run --release ...`